src/CodeExampleLanguageProcessing.ts

```diff
@@ -0,0 +1,371 @@
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === language
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else {
+    const maybeNodeLang = node.options?.language;
+    if (maybeNodeLang != undefined) {
+      const language = maybeNodeLang;
+      const mapping = languageValueMappings.find(
+        (mapper) => mapper.directiveValue === language
+      );
+      if (mapping != undefined) {
+        return mapping;
+      } else {
+        /* TODO: Maybe want to look at node.argument[0].value and try to parse the filepath/extension for language mapping
+         * Would need to add to the DirectiveNode an array of 'argument' with the following properties:
+         * type: string, position: object, value: string
+         * The 'value' field would contain the filepath with extension, similar to:
+         * value: '/includes/avs-examples/index-management/create-index/basic-example-coroutine.kt'
+         */
+        return {
+          directiveValue: "none",
+          canonicalValue: CanonicalLanguageValues.UNDEFINED,
+          extension: LanguageFileExtensions.UNDEFINED,
+        };
+      }
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  }
+};
+
+export const getLanguageDetailsForIoCodeBlock = (
+  node: DirectiveNode
+): LanguageMapper => {
+  const maybeNodeLanguage = node.options?.language;
+  const maybeChildNode = node.children[0];
+  if (maybeNodeLanguage != undefined) {
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue == (maybeNodeLanguage as string)
+    );
+    if (mapping != undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else if (maybeChildNode != undefined && (maybeChildNode as CodeNode)) {
+    const mapping = languageValueMappings.find(
+      (mapper) => mapper.directiveValue === maybeChildNode.lang
+    );
+    if (mapping !== undefined) {
+      return mapping;
+    } else {
+      return {
+        directiveValue: "none",
+        canonicalValue: CanonicalLanguageValues.UNDEFINED,
+        extension: LanguageFileExtensions.UNDEFINED,
+      };
+    }
+  } else {
+    return {
+      directiveValue: "none",
+      canonicalValue: CanonicalLanguageValues.UNDEFINED,
+      extension: LanguageFileExtensions.UNDEFINED,
+    };
+  }
+};
+
+export const getNewDefaultLangCounterMap = (): Map<string, number> => {
+  return new Map<string, number>([
+    [CanonicalLanguageValues.BASH, 0],
```

Comment:
Another good reason just to use a map: can keep this automatically up-to-date as new entries are added...

something like this (not tested)
```ts
Array.from(
  Object.values(directiveValueToLanguageInfo).reduce((uniqueNamesSet, value) => {
    uniqueNamesSet.add(value[0]);
    return uniqueNamesSet;
  }, new Set<string>() /* <-- initializer for `uniqueNamesSet` */)
).map((name) => ([name, 0]));
```

Otherwise bugs will exist in the future when someone forgets to update this function after adding a new language entry above.